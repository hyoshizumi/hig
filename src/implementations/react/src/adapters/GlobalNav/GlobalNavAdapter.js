import * as HIG from 'hig-vanilla';
import * as PropTypes from 'prop-types';

import HIGElement from '../../elements/HIGElement';
import HIGChildValidator from '../../elements/HIGChildValidator';
import createComponent from '../createComponent';

import SideNavAdapterComponent, { SideNavAdapter } from './SideNav/SideNavAdapter';
import TopNavAdapterComponent, {
  TopNavAdapter
} from './TopNav/TopNavAdapter';
import SubNavAdapterComponent, {
  SubNavAdapter
} from './SubNav/SubNavAdapter';
import Slot from '../../elements/components/GlobalNav/Slot';
import SideNav from '../../elements/components/GlobalNav/SideNav';

class GlobalNavAdapter extends HIGElement {
  constructor(initialProps) {
    super(HIG.GlobalNav, initialProps);
  }

  componentDidMount() {
    // Add any children
    if (this.sideNav) {
      this.hig.addSideNav(this.sideNav.hig);
      this.sideNav.mount();
    }

    if (this.topNav) {
      this.hig.addTopNav(this.topNav.hig);
      this.topNav.mount();
    }

    if (this.subNav) {
      this.hig.addSubNav(this.subNav.hig);
      this.subNav.mount();
    }

    if (this.initialProps.sideNavOpen) {
      this.hig.showSideNav();
    } else {
      this.hig.hideSideNav();
    }

    if (this.slot) {
      this.hig.addSlot(this.slot);
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case TopNavAdapter:
        return new TopNavAdapter(this.hig.partials.TopNav, props);
      case SubNavAdapter:
        return new SubNavAdapter(this.hig.partials.SubNav, props);
      case SideNavAdapter:
        return new SideNavAdapter(this.hig.partials.SideNav, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance, beforeChild = {}) {
    if (instance instanceof SideNavAdapter) {
      if (this.sideNav) {
        throw new Error('only one SideNav is allowed');
      } else {
        this.sideNav = instance;
        if (this.mounted) {
          this.hig.addSideNav(instance.hig);
          instance.mount();
        }
      }
    } else if (instance instanceof TopNavAdapter) {
      if (this.topNav) {
        throw new Error('only one TopNav is allowed');
      } else {
        this.topNav = instance;
        if (this.mounted) {
          this.hig.addTopNav(instance.hig);
          instance.mount();
        }
      }
    } else if (instance instanceof SubNavAdapter) {
      if (this.subNav) {
        throw new Error('only one SubNav is allowed');
      } else {
        this.subNav = instance;
        if (this.mounted) {
          this.hig.addSubNav(instance.hig);
          instance.mount();
        }
      }
    } else {
      throw new Error('unknown type');
    }
  }

  addSlot(element) {
    if (this.mounted) {
      this.hig.addSlot(element);
    } else {
      this.slot = element;
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    if (instance instanceof SideNavAdapter) {
      this.sideNav = null;
    } else if (instance instanceof TopNavAdapter) {
      this.topNav = null;
    } else if (instance instanceof SubNavAdapter) {
      this.subNav = null;
    }
    instance.unmount();
  }

  commitUpdate(updatePayload, oldProps, newProp) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'sideNavOpen': {
          propValue ? this.hig.showSideNav() : this.hig.hideSideNav();
          break;
        }
        case 'children': {
          // No-op
          break;
        }
        default: {
          console.warn(`${propKey} is unknown`);
        }
      }
    }
  }
}

const GlobalNavAdapterComponent = createComponent(GlobalNavAdapter);

GlobalNavAdapterComponent.propTypes = {
  sideNavOpen: PropTypes.bool,
  children: HIGChildValidator([
    SideNav,
    SideNavAdapterComponent,
    TopNavAdapterComponent,
    SubNavAdapterComponent,
    Slot,
  ])
};

GlobalNavAdapterComponent.__docgenInfo = {
  props: {
    sideNavOpen: {
      description: 'show or hide the SideNav'
    },

    children: {
      description: 'support adding SideNav, SubNav, or TopNav'
    }
  }
};

GlobalNavAdapterComponent.SideNav = SideNavAdapterComponent;
GlobalNavAdapterComponent.TopNav = TopNavAdapterComponent;
GlobalNavAdapterComponent.SubNav = SubNavAdapterComponent;
GlobalNavAdapterComponent.Slot = Slot;

export default GlobalNavAdapterComponent;
