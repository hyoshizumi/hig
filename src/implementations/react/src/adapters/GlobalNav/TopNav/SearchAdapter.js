/* Copyright 2016 Autodesk,Inc.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

 */

import * as PropTypes from 'prop-types';
import HIGElement from '../../../elements/HIGElement';
import createComponent from '../../createComponent';

export class SearchAdapter extends HIGElement {
  componentDidMount() {
    this.hig.onInput(this.topNavSearchOnInput.bind(this));
    this.hig.onClearIconClick(this.topNavClearInput.bind(this));
  }

  commitUpdate(updatePayload) {
    const mapping = {
      placeholder: 'setPlaceholder',
      query: 'setQuery'
    };
    super.commitUpdateWithMapping(updatePayload, mapping);
  }

  topNavSearchOnInput(event) {
    this.hig.showClearIcon();
    this.hig.setQuery(event.target.value);
  }

  topNavClearInput() {
    this.hig.setQuery('');
    this.hig.hideClearIcon();
  }
}

const SearchAdapterComponent = createComponent(SearchAdapter);

SearchAdapterComponent.propTypes = {
  placeholder: PropTypes.string,
  query: PropTypes.string,
  onClearIconClick: PropTypes.func,
  onInput: PropTypes.func,
  onFocusIn: PropTypes.func,
  onFocusOut: PropTypes.func
};

SearchAdapterComponent.__docgenInfo = {
  props: {
    placeholder: { description: '{String} sets the placeholder text' },
    query: { description: '{String} sets the search query text' },
    onClearIconClick: {
      description: '{function} Triggers when user clicks clear icon'
    },
    onInput: {
      description: '{function} Triggers when user enters text into search field'
    },
    onFocusIn: { description: '{function} Triggers when input has focus' },
    onFocusOut: { description: '{function} Triggers when input loses focus' }
  }
};
export default SearchAdapterComponent;
