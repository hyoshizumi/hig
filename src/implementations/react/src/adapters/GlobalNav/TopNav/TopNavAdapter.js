import * as PropTypes from 'prop-types';

import HIGElement from '../../../elements/HIGElement';
import HIGChildValidator from '../../../elements/HIGChildValidator';
import createComponent from '../../createComponent';

import HIGNodeList from '../../../elements/HIGNodeList';
import ProjectAccountSwitcherComponent, {
  ProjectAccountSwitcherAdapter
} from './ProjectAccountSwitcherAdapter';
import ProjectAccountSwitcher
  from '../../../elements/components/GlobalNav/TopNav/ProjectAccountSwitcher';
import ProfileComponent, {
  ProfileAdapter
} from './ProfileAdapter';
import ShortcutAdapterComponent, { ShortcutAdapter } from './ShortcutAdapter';
import HelpComponent, { HelpAdapter } from './HelpAdapter';
import SearchComponent, {
  SearchAdapter
} from './SearchAdapter';

export class TopNavAdapter extends HIGElement {
  constructor(HIGConstructor, initialProps) {
    super(HIGConstructor, initialProps);

    this.shortcuts = new HIGNodeList({
      ShortcutAdapter: {
        type: ShortcutAdapter,
        HIGConstructor: this.hig.partials.Shortcut,
        onAdd: (instance, beforeInstance) => {
          this.hig.addShortcut(instance, beforeInstance);
        }
      }
    });
  }

  componentDidMount() {
    // Add any children
    if (this.profile) {
      this.hig.addProfile(this.profile.hig);
      this.profile.mount();
    }

    if (this.projectAccountSwitcher) {
      this.hig.addProjectAccountSwitcher(this.projectAccountSwitcher.hig);
      this.projectAccountSwitcher.mount();
    }

    this.shortcuts.componentDidMount();

    if (this.help) {
      this.hig.addHelp(this.help.hig);
      this.help.mount();
    }

    if (this.search) {
      this.hig.addSearch(this.search.hig);
      this.search.mount();
    }
  }

  commitUpdate(updatePayload) {
    for (let i = 0; i < updatePayload.length; i += 2) {
      const propKey = updatePayload[i];
      const propValue = updatePayload[i + 1];

      switch (propKey) {
        case 'children': {
          // No-op
          break;
        }
        case 'logo': {
          this.hig.setLogo(propValue);
          break;
        }
        case 'logoLink': {
          this.hig.setLogoLink(propValue);
          break;
        }
        case 'onLogoClick': {
          const dispose = this._disposeFunctions.get('onLogoClickDispose');

          if (dispose) {
            dispose();
          }

          this._disposeFunctions.set(
            'onLogoClickDispose',
            this.hig.onLogoClick(propValue)
          );
          break;
        }
        default: {
          // No-op
        }
      }
    }
  }

  createElement(ElementConstructor, props) {
    switch (ElementConstructor) {
      case ProfileAdapter:
        return new ProfileAdapter(this.hig.partials.Profile, props);
      case ProjectAccountSwitcherAdapter:
        return new ProjectAccountSwitcherAdapter(
          this.hig.partials.ProjectAccountSwitcher,
          props
        );
      case HelpAdapter:
        return new HelpAdapter(this.hig.partials.Help, props);
      case ShortcutAdapter:
        return this.shortcuts.createElement(ElementConstructor, props);
      case SearchAdapter:
        return new SearchAdapter(this.hig.partials.Search, props);
      default:
        throw new Error(`Unknown type ${ElementConstructor.name}`);
    }
  }

  appendChild(instance) {
    this.requireSingleInstance(instance);
    this.checkValidChild(instance);

    if (instance instanceof ShortcutAdapter) {
      this.shortcuts.insertBefore(instance);
    } else {
      this[this.getPropertyNameFor(instance)] = instance;
      if (this.mounted) {
        this.hig[this.getFunctionNameFor(instance)](instance.hig);
        instance.mount();
      }
    }
  }

  insertBefore(instance) {
    this.appendChild(instance);
  }

  removeChild(instance) {
    if (instance instanceof ProfileAdapter) {
      this.profile = null;
    }

    instance.unmount();
  }

  requireSingleInstance(instance) {
    const requiredSingle = [
      'Profile',
      'ProjectAccountSwitcher',
      'HelpAdapter',
      'SearchAdapter'
    ];
    super.requireSingleInstance(instance, requiredSingle);
  }

  checkValidChild(instance) {
    const validInstances = [
      ProfileAdapter,
      ProjectAccountSwitcherAdapter,
      ShortcutAdapter,
      HelpAdapter,
      SearchAdapter
    ];
    if (!validInstances.includes(instance.constructor)) {
      throw new Error(
        `${instance.constructor.name} is not a valid child element of this parent.`
      );
    }
  }

  getPropertyNameFor(instance) {
    if (instance instanceof ProfileAdapter) {
      return 'profile';
    }
    if (instance instanceof ProjectAccountSwitcherAdapter) {
      return 'projectAccountSwitcher';
    }
    if (instance instanceof ShortcutAdapter) {
      return 'shortcut';
    }
    if (instance instanceof HelpAdapter) {
      return 'help';
    }
    if (instance instanceof SearchAdapter) {
      return 'search';
    }
    return null;
  }

  getFunctionNameFor(instance) {
    return `add${instance.constructor.name}`;
  }

  onHamburgerClick(callback) {
    this.hig.onHamburgerClick(callback);
  }
}

const TopNavAdapterComponent = createComponent(TopNavAdapter);

TopNavAdapterComponent.propTypes = {
  logo: PropTypes.string,
  logoLink: PropTypes.string,
  onLogoClick: PropTypes.func,
  onHamburgerClick: PropTypes.func,
  children: HIGChildValidator([
    ProfileComponent,
    ProjectAccountSwitcherComponent,
    ProjectAccountSwitcher,
    ShortcutAdapterComponent,
    HelpComponent,
    SearchComponent
  ])
};

TopNavAdapterComponent.__docgenInfo = {
  props: {
    logo: {
      description: 'sets the logo'
    },
    logoLink: {
      description: 'sets the link of the logo'
    },
    onHamburgerClick: {
      description: 'triggered when hamburger icon is clicked'
    },
    onLogoClick: {
      description: 'triggered when logo is clicked'
    }
  }
};

TopNavAdapterComponent.Profile = ProfileComponent;
TopNavAdapterComponent.Shortcut = ShortcutAdapterComponent;
TopNavAdapterComponent.Help = HelpComponent;
TopNavAdapterComponent.ProjectAccountSwitcher = ProjectAccountSwitcherComponent;
TopNavAdapterComponent.Search = SearchComponent;

export default TopNavAdapterComponent;
