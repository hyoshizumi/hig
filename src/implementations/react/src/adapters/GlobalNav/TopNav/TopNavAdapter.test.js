/* eslint-disable react/prop-types, no-console */
import { mount } from 'enzyme';
import * as HIG from 'hig-vanilla';
import React from 'react';

import GlobalNavAdapter from '../GlobalNavAdapter';
import TopNavAdapter from './TopNavAdapter';
import ProfileAdapter from './ProfileAdapter';
import SearchAdapter from './SearchAdapter';

const Context = props => (
  <GlobalNavAdapter>
    <TopNavAdapter>
      <SearchAdapter
        placeholder={props.placeholder}
        query={props.query}
        onInput={props.onInput}
        onFocusIn={props.onFocusIn}
        onFocusOut={props.onFocusOut}
      />
    </TopNavAdapter>
  </GlobalNavAdapter>
);

describe('<TopNav>', () => {
  // Create the GlobalNav context for the TopNav to be attached to
  const defaults = {
    logo: '../../../bim-logo.png',
    logoLink: 'http://www.autodesk.com'
  };

  const profileDefaults = { image: '../../../bim-logo.png' };
  function createHigNav() {
    const domContainer = document.createElement('div');
    const higNav = new HIG.GlobalNav();
    higNav.mount(domContainer);

    return { higNav, domContainer };
  }

  function createTopNav() {
    const { higNav, domContainer } = createHigNav();
    const topNav = new higNav.partials.TopNav({ ...defaults });
    higNav.addTopNav(topNav);
    return { topNav, higNav, domContainer };
  }

  it('renders a topnav', () => {
    const reactContainer = document.createElement('div');

    mount(
      <GlobalNavAdapter>
        <TopNavAdapter {...defaults}>
          <ProfileAdapter {...profileDefaults} />
        </TopNavAdapter>
      </GlobalNavAdapter>,
      { attachTo: reactContainer }
    );
    expect(reactContainer.firstElementChild.outerHTML).toMatchSnapshot();
  });

  it('can render a Profile', () => {
    const { topNav, higNav, domContainer } = createTopNav();

    const profile = new topNav.partials.Profile({ ...profileDefaults });
    topNav.addProfile(profile);

    const reactContainer = document.createElement('div');
    mount(
      <GlobalNavAdapter>
        <TopNavAdapter {...defaults}>
          <ProfileAdapter {...profileDefaults} />
        </TopNavAdapter>
      </GlobalNavAdapter>,
      { attachTo: reactContainer }
    );

    expect(reactContainer.firstElementChild.outerHTML).toEqual(
      domContainer.firstElementChild.outerHTML
    );
  });

  function anyHandler() {
    return true;
  }

  it('can render Search programmatically', () => {
    const { topNav, domContainer } = createTopNav();
    const props = {
      placeholder: 'enter some text',
      query: 'foobar',
      onInput: anyHandler,
      onFocusIn: anyHandler,
      onFocusOut: anyHandler
    };
    const tnsearch = new topNav.partials.Search(props);
    topNav.addSearch(tnsearch);

    expect(domContainer.firstChild.outerHTML).toMatchSnapshot();
    const elems = domContainer.getElementsByClassName(
      'hig__global-nav__top-nav__search__inputholder__input'
    );
    expect(elems.length).toEqual(1);
  });

  it('can render Search through React-like components', () => {
    const reactContainer = document.createElement('div');
    const props = {
      placeholder: 'enter some text',
      query: 'foobar',
      onInput: anyHandler,
      onFocusIn: anyHandler,
      onFocusOut: anyHandler
    };

    mount(Context(props), { attachTo: reactContainer });
    expect(reactContainer.firstChild.outerHTML).toMatchSnapshot();

    const elems = reactContainer.getElementsByClassName(
      'hig__global-nav__top-nav__search__inputholder__input'
    );
    expect(elems.length).toEqual(1);
  });
});
