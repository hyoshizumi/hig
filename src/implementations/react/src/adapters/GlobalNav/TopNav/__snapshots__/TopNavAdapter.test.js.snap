// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNav> can render Search programmatically 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <div class=\\"hig__global-nav__top-nav\\">
  <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--left\\">
    <div class=\\"hig__global-nav__top-nav__hamburger\\">
      <div class=\\"hig__global-nav__top-nav__hamburger__hamburgericon\\"><div class=\\"hig__icon\\"><svg width=\\"16\\" height=\\"12\\" viewBox=\\"0 0 16 12\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"hamburger\\"><g transform=\\"translate(-4 -6)\\" fill=\\"#353636\\" fill-rule=\\"evenodd\\"><rect x=\\"4\\" y=\\"6\\" width=\\"16\\" height=\\"2\\"></rect><rect x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect><rect x=\\"4\\" y=\\"16\\" width=\\"16\\" height=\\"2\\"></rect></g></svg></div></div>
      <div class=\\"hig__global-nav__top-nav__hamburger__closeicon\\"><div class=\\"hig__icon\\"><svg width=\\"18\\" height=\\"18\\" viewBox=\\"0 0 18 18\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-hamburger\\"><g stroke=\\"#0A131C\\" stroke-width=\\"2\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"square\\"><path d=\\"M18.754628,5.24537196 L5.24537196,18.754628\\" transform=\\"translate(-3 -3)\\"></path><path d=\\"M5.24537196,5.24537196 L18.754628,18.754628\\" transform=\\"translate(-3 -3)\\"></path></g></svg></div></div>
    </div>
    <div class=\\"hig__global-nav__top-nav__logo\\">
      <a href=\\"http://www.autodesk.com\\">
        <img src=\\"../../../bim-logo.png\\">
      </a>
    </div>
  </div>
  <div class=\\"hig__global-nav__top-nav__search\\">
    <div class=\\"hig__global-nav__top-nav__search__icon\\"><div class=\\"hig__icon\\"><svg width=\\"22\\" height=\\"22\\" viewBox=\\"0 0 22 22\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"search\\"><g transform=\\"translate(-1 -1)\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><rect fill=\\"#353636\\" transform=\\"rotate(-45 19.196 18.891)\\" x=\\"18.195\\" y=\\"13.891\\" width=\\"2.001\\" height=\\"10\\" rx=\\"1\\"></rect><circle stroke=\\"#353636\\" stroke-width=\\"1.5\\" cx=\\"10\\" cy=\\"10\\" r=\\"8\\"></circle></g></svg></div></div>
    <div class=\\"hig__global-nav__top-nav__search__inputholder\\"><input class=\\"hig__global-nav__top-nav__search__inputholder__input\\" type=\\"text\\" placeholder=\\"enter some text\\" value=\\"foobar\\"></div>
    <div class=\\"hig__global-nav__top-nav__search__clear\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-small\\"><title>hig/icon/close-small</title><g fill=\\"none\\" fill-rule=\\"evenodd\\"><circle stroke=\\"#353636\\" cx=\\"12\\" cy=\\"12\\" r=\\"8\\"></circle><polygon fill=\\"#353636\\" transform=\\"rotate(45 12 12)\\" points=\\"16.5 11.4518414 12.5847458 11.4518414 12.5847458 7.5 11.4152542 7.5 11.4152542 11.4518414 7.5 11.4518414 7.5 12.5226629 11.4152542 12.5226629 11.4152542 16.5 12.5847458 16.5 12.5847458 12.5226629 16.5 12.5226629\\"></polygon></g></svg></div></div>
</div><!--SEARCH-->
  <div class=\\"hig__global-nav__top-nav__spacer\\"></div>
  <!--PROJECT_ACCOUNT_SWITCHER-->
  <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
  <!--SHORTCUT-->
  </div>
  <!--PROFILE-->
</div><!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<TopNav> can render Search through React-like components 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <div class=\\"hig__global-nav__top-nav\\">
  <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--left\\">
    <div class=\\"hig__global-nav__top-nav__hamburger\\">
      <div class=\\"hig__global-nav__top-nav__hamburger__hamburgericon\\"><div class=\\"hig__icon\\"><svg width=\\"16\\" height=\\"12\\" viewBox=\\"0 0 16 12\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"hamburger\\"><g transform=\\"translate(-4 -6)\\" fill=\\"#353636\\" fill-rule=\\"evenodd\\"><rect x=\\"4\\" y=\\"6\\" width=\\"16\\" height=\\"2\\"></rect><rect x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect><rect x=\\"4\\" y=\\"16\\" width=\\"16\\" height=\\"2\\"></rect></g></svg></div></div>
      <div class=\\"hig__global-nav__top-nav__hamburger__closeicon\\"><div class=\\"hig__icon\\"><svg width=\\"18\\" height=\\"18\\" viewBox=\\"0 0 18 18\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-hamburger\\"><g stroke=\\"#0A131C\\" stroke-width=\\"2\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"square\\"><path d=\\"M18.754628,5.24537196 L5.24537196,18.754628\\" transform=\\"translate(-3 -3)\\"></path><path d=\\"M5.24537196,5.24537196 L18.754628,18.754628\\" transform=\\"translate(-3 -3)\\"></path></g></svg></div></div>
    </div>
    <div class=\\"hig__global-nav__top-nav__logo\\">
      <a>
        <img src=\\"\\">
      </a>
    </div>
  </div>
  <div class=\\"hig__global-nav__top-nav__search\\">
    <div class=\\"hig__global-nav__top-nav__search__icon\\"><div class=\\"hig__icon\\"><svg width=\\"22\\" height=\\"22\\" viewBox=\\"0 0 22 22\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"search\\"><g transform=\\"translate(-1 -1)\\" fill=\\"none\\" fill-rule=\\"evenodd\\"><rect fill=\\"#353636\\" transform=\\"rotate(-45 19.196 18.891)\\" x=\\"18.195\\" y=\\"13.891\\" width=\\"2.001\\" height=\\"10\\" rx=\\"1\\"></rect><circle stroke=\\"#353636\\" stroke-width=\\"1.5\\" cx=\\"10\\" cy=\\"10\\" r=\\"8\\"></circle></g></svg></div></div>
    <div class=\\"hig__global-nav__top-nav__search__inputholder\\"><input class=\\"hig__global-nav__top-nav__search__inputholder__input\\" type=\\"text\\" placeholder=\\"enter some text\\" value=\\"foobar\\"></div>
    <div class=\\"hig__global-nav__top-nav__search__clear\\"><div class=\\"hig__icon\\"><svg width=\\"24\\" height=\\"24\\" viewBox=\\"0 0 24 24\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-small\\"><title>hig/icon/close-small</title><g fill=\\"none\\" fill-rule=\\"evenodd\\"><circle stroke=\\"#353636\\" cx=\\"12\\" cy=\\"12\\" r=\\"8\\"></circle><polygon fill=\\"#353636\\" transform=\\"rotate(45 12 12)\\" points=\\"16.5 11.4518414 12.5847458 11.4518414 12.5847458 7.5 11.4152542 7.5 11.4152542 11.4518414 7.5 11.4518414 7.5 12.5226629 11.4152542 12.5226629 11.4152542 16.5 12.5847458 16.5 12.5847458 12.5226629 16.5 12.5226629\\"></polygon></g></svg></div></div>
</div><!--SEARCH-->
  <div class=\\"hig__global-nav__top-nav__spacer\\"></div>
  <!--PROJECT_ACCOUNT_SWITCHER-->
  <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
  <!--SHORTCUT-->
  </div>
  <!--PROFILE-->
</div><!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;

exports[`<TopNav> renders a topnav 1`] = `
"<div class=\\"hig__global-nav\\">
    <!--SIDENAV-->
    <div class=\\"hig__global-nav__container\\">
        <div class=\\"hig__global-nav__top-nav\\">
  <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--left\\">
    <div class=\\"hig__global-nav__top-nav__hamburger\\">
      <div class=\\"hig__global-nav__top-nav__hamburger__hamburgericon\\"><div class=\\"hig__icon\\"><svg width=\\"16\\" height=\\"12\\" viewBox=\\"0 0 16 12\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"hamburger\\"><g transform=\\"translate(-4 -6)\\" fill=\\"#353636\\" fill-rule=\\"evenodd\\"><rect x=\\"4\\" y=\\"6\\" width=\\"16\\" height=\\"2\\"></rect><rect x=\\"4\\" y=\\"11\\" width=\\"16\\" height=\\"2\\"></rect><rect x=\\"4\\" y=\\"16\\" width=\\"16\\" height=\\"2\\"></rect></g></svg></div></div>
      <div class=\\"hig__global-nav__top-nav__hamburger__closeicon\\"><div class=\\"hig__icon\\"><svg width=\\"18\\" height=\\"18\\" viewBox=\\"0 0 18 18\\" xmlns=\\"http://www.w3.org/2000/svg\\" role=\\"img\\" title=\\"close-hamburger\\"><g stroke=\\"#0A131C\\" stroke-width=\\"2\\" fill=\\"none\\" fill-rule=\\"evenodd\\" stroke-linecap=\\"square\\"><path d=\\"M18.754628,5.24537196 L5.24537196,18.754628\\" transform=\\"translate(-3 -3)\\"></path><path d=\\"M5.24537196,5.24537196 L18.754628,18.754628\\" transform=\\"translate(-3 -3)\\"></path></g></svg></div></div>
    </div>
    <div class=\\"hig__global-nav__top-nav__logo\\">
      <a href=\\"http://www.autodesk.com\\">
        <img src=\\"../../../bim-logo.png\\">
      </a>
    </div>
  </div>
  <!--SEARCH-->
  <div class=\\"hig__global-nav__top-nav__spacer\\"></div>
  <!--PROJECT_ACCOUNT_SWITCHER-->
  <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
  <!--SHORTCUT-->
  </div>
  <div class=\\"hig__global-nav__profile\\">
    <div class=\\"hig__global-nav__top-nav__group hig__global-nav__top-nav__group--right\\">
        <!--FLYOUT-->
    <div class=\\"hig__flyout hig__flyout--anchor-top-right\\">
    <div class=\\"hig__global-nav__top-nav__profile__profile-image\\">
    <img class=\\"hig__global-nav__top-nav__profile__profile-image__image\\" src=\\"../../../bim-logo.png\\">
</div><!--TARGET-->
    <div class=\\"hig__flyout__container\\">
        <div class=\\"hig__flyout__chevron\\"></div>
        <div class=\\"hig__flyout__panel\\">
            <div class=\\"hig__global-nav__top-nav__profile__profile-flyout-content\\">
    <div class=\\"hig__global-nav__top-nav__profile__profile-flyout-content__name\\">name</div>
    <div class=\\"hig__global-nav__top-nav__profile__profile-flyout-content__email\\">email@example.com</div>
    <div class=\\"hig__global-nav__top-nav__profile__profile-flyout-content__links\\">
        <a href=\\"#\\" tabindex=\\"0\\" class=\\"hig__button hig__button--secondary hig__button--small\\" title=\\"Sign Out\\">
    <!--ICON-->
    <span class=\\"hig__button__title\\">Sign Out</span>
</a><!--SIGN_OUT_BUTTON-->
        <a href=\\"https://www.autodesk.com\\" tabindex=\\"0\\" class=\\"hig__button hig__button--secondary hig__button--small\\" title=\\"Profile Settings\\">
    <!--ICON-->
    <span class=\\"hig__button__title\\">Profile Settings</span>
</a><!--SETTINGS_LINK-->
    </div>
</div><!--SLOT-->
        </div>
    </div>
</div></div>
</div><!--PROFILE-->
</div><!--TOPNAV-->
        <!--SUBNAV-->
        <div class=\\"hig__global-nav__slot\\">
            <!--SLOT-->
        </div>
    </div>
</div>"
`;
